pkg/operation/botanist/secrets.go

```diff
@@ -1222,3 +401,7 @@ func dnsNamesForEtcd(namespace string) []string {
 	names = append(names, dnsNamesForService(fmt.Sprintf("%s-client", v1beta1constants.ETCDEvents), namespace)...)
 	return names
 }
+
+func shouldRotateBasicAuthCredentials(shootWantsBasicAuth, shootHasBasicAuth bool) bool {
+	return !shootWantsBasicAuth && shootHasBasicAuth || shootWantsBasicAuth && !shootHasBasicAuth
+}
```

Comment:
```suggestion
func shouldRotateBasicAuthCredentials(shootWantsBasicAuth, shootHasBasicAuth bool) bool {
	return shootWantsBasicAuth != shootHasBasicAuth
}
```
Given this simplification and the confusing function name we probably don't need the function at all.